tests/unit/lms/validation/authentication/_launch_params_test.py

```diff
@@ -1,18 +1,38 @@
 import pytest
 from pyramid.httpexceptions import HTTPUnprocessableEntity
 
-from lms.models import LTIUser
 from lms.services import LTILaunchVerificationError
 from lms.validation import ValidationError
 from lms.validation.authentication import LaunchParamsAuthSchema
 
 
 class TestLaunchParamsAuthSchema:
-    def test_it_returns_the_lti_user_info(self, schema):
+    def test_it_returns_the_lti_user_info(self, schema, LTIUser):
         lti_user = schema.lti_user()
 
-        assert lti_user == LTIUser(
-            "TEST_USER_ID", "TEST_OAUTH_CONSUMER_KEY", "TEST_ROLES"
+        LTIUser.from_names.assert_called_once_with(
+            "TEST_GIVEN_NAME",
+            "TEST_FAMILY_NAME",
+            "TEST_FULL_NAME",
+            user_id="TEST_USER_ID",
+            oauth_consumer_key="TEST_OAUTH_CONSUMER_KEY",
+            roles="TEST_ROLES",
+            tool_consumer_instance_guid="TEST_TOOL_CONSUMER_INSTANCE_GUID",
+        )
+        assert lti_user == LTIUser.from_names.return_value
```

Comment:
Previously `LTIUser` was just a simple value class, but now its grown a non-trivial factory method (which calls a non-trivial private collaborator) so I decided to patch `LTIUser` in `LaunchParamsAuthSchema`'s unit tests. Some reasoning as for patching `display_name()` in `LTIUser`s unit tests

Reply:
This is another point where adding the factory method has imposed a cost. With the other approach the LTIUser object would remain a plain value object.